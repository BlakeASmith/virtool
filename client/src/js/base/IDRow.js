import React from "react";
import { connect } from "react-redux";

const IDRowComponent = ({ id, showIds }) => {
    if (showIds) {
        return (
            <tr>
                <th>Unique ID</th>
                <td>{id}</td>
            </tr>
        );
    }

    return null;
};

const mapStateToProps = (state) => ({
    showIds: state.account.settings.show_ids
});

export const IDRow = connect(mapStateToProps)(IDRowComponent);
