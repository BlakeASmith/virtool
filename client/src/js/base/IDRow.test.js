import { IDRowComponent } from "./IDRow";

describe("<IDRow />", () => {
    let props;
    let wrapper;

    it("when [account.settings.show_ids=false], renders nothing (null)", () => {
        props = { id: "test_id", showIds: false };
        wrapper = shallow(<IDRowComponent {...props} />);

        expect(wrapper.html()).toBe(null);
        expect(wrapper).toMatchSnapshot();
    });

    describe("when [account.settings.show_ids=true]: ", () => {
        
        beforeEach(() => {
            props = { id: "test_id", showIds: true };
            wrapper = mount(
                <table>
                    <tbody>
                        <IDRowComponent {...props} />
                    </tbody>
                </table>
            );
        });

        it("renders correctly", () => {
            expect(wrapper).toMatchSnapshot();
        });

        it("renders a table row with a header and the target's unique database id", () => {
            
            expect(wrapper.find('tr').exists()).toBe(true);
            expect(wrapper.find('th').text()).toEqual("Unique ID");
            expect(wrapper.find('td').text()).toEqual(props.id);
        });

    });

});
