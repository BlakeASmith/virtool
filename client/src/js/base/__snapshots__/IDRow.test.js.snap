// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IDRow /> when [account.settings.show_ids=false], renders nothing (null) 1`] = `""`;

exports[`<IDRow /> when [account.settings.show_ids=true]:  renders correctly 1`] = `
<table>
  <tbody>
    <IDRowComponent
      id="test_id"
      showIds={true}
    >
      <tr>
        <th>
          Unique ID
        </th>
        <td>
          test_id
        </td>
      </tr>
    </IDRowComponent>
  </tbody>
</table>
`;
